### DISCUSSION TITLE
About H264 and H265 encoding options disappearing

### DISCUSSION THREAD

**Discussion by MrChenhtlss** [rustdesk/rustdesk#6512] [Q&A] [answered]
H265 and H264 encoding options are not available in versions 1.2.3 and 1.2.4. I don’t know if it is no longer available, or if this option has disappeared. Can you tell me how to enable it?
![20231123](https://github.com/rustdesk/rustdesk/assets/146620104/22b4a58d-6bb2-481b-bde3-2e9072ecca01)
![20231122](https://github.com/rustdesk/rustdesk/assets/146620104/7572ca71-11fb-450b-8b6f-2291c50078d5)

**Comment by rustdesk** [comment]
@21pages

**Reply by MrChenhtlss** [reply]
why is it like this? Is it a problem with my operation or can it be solved?

**Reply by 21pages** [reply]
This issue may occur when a connection is established when the application is just started because the application checks for hardware codec availability at startup. Otherwise, it should not be supported, please check `%appdata%\RustDesk\config\RustDesk_hwcodec.toml` or `C:\Windows\ServiceProfiles\LocalService\AppData\Roaming\RustDesk\config\RustDesk_hwcodec.toml`. And only codecs supported by both sides will be displayed.

**Reply by MrChenhtlss** [reply]
`[options]
bestHwEncoders = '''
{
  "h264": {
    "name": "h264_qsv",
    "format": "H264",
    "vendor": "INTEL",
    "score": 70,
    "hwdevice": "AV_HWDEVICE_TYPE_NONE"
  },
  "h265": {
    "name": "hevc_qsv",
    "format": "H265",
    "vendor": "INTEL",
    "score": 60,
    "hwdevice": "AV_HWDEVICE_TYPE_NONE"
  }
}'''
bestHwDecoders = '''
{
  "h264": {
    "name": "h264",
    "format": "H264",
    "vendor": "OTHER",
    "score": 91,
    "hwdevice": "AV_HWDEVICE_TYPE_D3D11VA"
  },
  "h265": {
    "name": "hevc",
    "format": "H265",
    "vendor": "OTHER",
    "score": 91,
    "hwdevice": "AV_HWDEVICE_TYPE_D3D11VA"
  }
}'''
`

**Reply by 21pages** [reply]
What about the other side?

**Reply by 21pages** [reply]
is the option checked both sides?

![1701161912210](https://github.com/rustdesk/rustdesk/assets/14891774/cfdaa014-d9c9-4d27-9d27-315b4c37f430)

**Reply by MrChenhtlss** [reply]
no

**Reply by 21pages** [reply]
Check the option and retry

**Reply by MrChenhtlss** [reply]
Do I need to turn this option on?

**Reply by 21pages** [reply, marked as answer]
Yes. if it works, could you try https://github.com/rustdesk/rustdesk/discussions/5264 to see whether https://github.com/oneapi-src/oneVPL-intel-gpu/issues/308 happens on your computer ?

**Reply by MrChenhtlss** [reply]
Problem found. The reason is that I don't have a separate nvidia graphics card and my CPU is not among the hardware platforms supported by the Intel(R) Media SDK GPU Runtime
in range

**Reply by 21pages** [reply]
That range is not a support range and I've deleted it, new hardware platforms still support mfx. Did the old hardware codec work after turning this option on?

**Reply by MrChenhtlss** [reply]
The old decoder status is normal. My controlled terminal is Intel 4th and the controlling terminal is Intel 8th. I think this is Intel fourth. This is an unsupported issue. I borrowed another Intel 8th PC and the launch controller can support H264 and H265.

**Reply by MrChenhtlss** [reply]
But I found another problem. The console is Intel 8th and does not have a discrete graphics card. The controlled terminal is Intel Windows Server 2019 but cannot use H265 H264.
![0255238a4efd42463cec11eaa7d37a8](https://github.com/rustdesk/rustdesk/assets/146620104/1fd008fc-a4b4-4dc6-af3c-e8b300515418)
![2b55fd2300f6d86e32fc7f6839a0565](https://github.com/rustdesk/rustdesk/assets/146620104/1801901f-9fa4-48e8-9b20-d67833f0943b)

**Reply by 21pages** [reply]
Did you check https://github.com/rustdesk/rustdesk/discussions/6512#discussioncomment-7689712 both sides?

**Reply by MrChenhtlss** [reply]
Yes, I confirmed it, and also updated the driver of the discrete graphics card and restarted the server.

**Reply by 21pages** [reply]
Could you show the content of `RustDesk_hwcodec.toml` of the controlled side? That image only means having the ability to decode.

**Reply by MrChenhtlss** [reply]
![a682c5533b670d6c0ecb06da34b2f5c](https://github.com/rustdesk/rustdesk/assets/146620104/3872c50c-8344-4a20-90db-dff15230e0c0)

**Reply by 21pages** [reply]
The controlled side doesn't have encoding ability. Please try https://github.com/rustdesk/rustdesk/discussions/5264

**Reply by MrChenhtlss** [reply]
Well thank you very much for your support
